ASM-4988 Destroy interface from all vLANS

logankimmel opened the pull request:
Added ensure => absent to remove the interface from all
switch vLAN's

logankimmel commented on lib/puppet/provider/force10_interface/dell_ftos.rb:
```diff
@@ -17,4 +17,69 @@ def flush
+        'fortygigabit' => 'fortyGigE',
+        'gigabit' => 'gigabitethernet'
+    }
+    # binding.pry
```
I'll remove this

gavin-scott commented on lib/puppet/provider/force10_interface/dell_ftos.rb:
```diff
@@ -17,4 +17,69 @@ def flush
+  end
+
+  def create
+    #Does nothing right now
```
How does this interact with the pre-existing configuration behavior that allowed setting e.g. mtu, portfast etc.? I assume that with `ensure = present` those values will still be set even though `create` does nothing? 

What about if a resource has `ensource = absent` and also specifies values for mtu, portfast, etc.? Is the  behavior that the VLANs are removed and mtu and portfast are set to the specified values?

logankimmel replied:
@gavin-scott That is correct.  On `ensure => present`, the interface is created with the settings specified, just as before.  If you were to use `ensure => absent` witch other interface config settings as well.  The interface will still get configured with these settings but will not be on any vlan
The site.pp that I tested with was such:
```puppet
node default {
asm::mxl { '18':
  vlan_name => '',
  untagged_tengigabitethernet => '0/1',
  tagged_portchannel => '',
  require => 'Force10_interface[Tengigabitethernet0/1]',
}
force10_interface { 'Tengigabitethernet0/1':
  switchport => 'true',
  shutdown => 'false',
  portmode => 'hybrid',
  portfast => 'portfast',
  edge_port => 'edge-port',
  mtu => '12000',
  protocol => 'lldp',
  ensure => absent,
}
}
```

gavin-scott commented:
I'd love to see some unit tests on this. It looks like there is an existing unit test for this type that @asaenz added.

logankimmel commented:
@gavin-scott this is G2G

ripienaar commented on lib/puppet_x/force10/model/switch.rb:
```diff
@@ -47,6 +47,12 @@ def skip_params_to_hash
 
+  def all_vlans
+   # binding.pry
+    grp = PuppetX::Force10::Model.const_get('vlan'.capitalize).new(transport, facts, {:name => 'vlan'})
```
I am probably missing something obvious but why the ```const_get```? that tends to be used only when you have a variable class name?

ripienaar replied:
just to be clear, from what I can see the above is identical to:

```
grp = PuppetX::Force10::Model::Vlan.new(transport, facts, {:name => 'vlan'})
```

Should also probably kill the commented out ```binding.pry``` there

ripienaar commented:
We should probably do this in a 2nd PR but we need similar handling for ```force10_fcoemap``` and ```force10_zoneset```

logankimmel commented:
@gavin-scott @ripienaar  I added support for s-series by moving the vlan_information fact into the base possible_facts.  This way the same logic can be used for removing the port interface from the vlans.

Added tests for s_series parsing of interfaces to remove

gavin-scott commented:
I think this is good to go, I'm going to go ahead and merge. @ripienaar note that @logankimmel refactored some of the facts to be more consistent between IOM and ToR -- it might simplify the switch config code a bit to adjust to that change at some point. He left the old facts in place so that nothing breaks.
